packages/lesswrong/lib/collections/chapters/callbacks.js

```diff
@@ -30,3 +32,38 @@ function ChaptersEditHTMLSerializeCallback (modifier, chapter) {
 }
 
 addCallback("chapters.edit.sync", ChaptersEditHTMLSerializeCallback);
+
+async function ChaptersEditCanonizeCallback (chapter) {
+  const posts = await Sequences.getAllPosts(chapter.sequenceId)
+  const sequence = await Sequences.find({_id:chapter.sequenceId}).fetch()[0]
+
+  _.range(posts.length).forEach((i) => {
+    
+    const validSequenceId = (currentPost, sequence) => {
+      // Only update a post if it either doesn't have a canonicalSequence, or if we're editing
+      // chapters *from* its canonicalSequence
+      return !currentPost.canonicalSequenceId || currentPost.canonicalSequenceId === sequence._id
+    }
+
+    const currentPost = posts[i]
+    if ((currentPost.userId === sequence.userId) && validSequenceId(currentPost, sequence)) {
+      let prevPost = {slug:""}
+      let nextPost = {slug:""}
+      if (i-1>=0) {
+        prevPost = posts[i-1]
+      }
+      if (i+1<posts.length) {
+        nextPost = posts[i+1]
+      }
+      Posts.update({slug: currentPost.slug}, {$set: {
```

Comment:
We only want to allow the author of a post to set the canonical sequence Id. Just because you are the first to create a sequence with my post in it, shouldn't mean that you get to set its canonical sequence.

Reply:
this should be included in the current code, no? (it checks if the post.userId matches the sequence userId

Reply:
So, this is a bit weird, because it can mean that you can have a single post in a sequence that you can navigate to, but can't navigate away from via the sequences navigation. 

I would recommend that we check that *all* posts are part of the relevant chapter.